lib/Conversion/MooreToCore/MooreToCore.cpp

```diff
@@ -160,17 +160,12 @@ struct BoolCastOpConversion : public OpConversionPattern<BoolCastOp> {
   LogicalResult
   matchAndRewrite(BoolCastOp op, OpAdaptor adaptor,
                   ConversionPatternRewriter &rewriter) const override {
-    if (cast<UnpackedType>(op.getInput().getType())
-            .castToSimpleBitVectorOrNull()) {
-      Type resultType = typeConverter->convertType(op.getInput().getType());
-      Value zero = rewriter.create<hw::ConstantOp>(op->getLoc(), resultType, 0);
-
-      rewriter.replaceOpWithNewOp<comb::ICmpOp>(op, comb::ICmpPredicate::ne,
-                                                adaptor.getInput(), zero);
-      return success();
-    }
+    Type resultType = typeConverter->convertType(op.getInput().getType());
```

Comment:
BoolCastOp does not guarantee that the input type is a simple bitvector, right? Thus `resultType` could be null and should be checked.

Reply:
Yes you're totally right. This should keep the old check.